src/libs/ajax.js

```diff
@@ -792,6 +804,19 @@ const Workspaces = signal => ({
 })
 
 
+const DataRepoSnapshots = signal => ({
+  snapshots: () => {
+    const root = `repository/v1/snapshots`
+
+    return {
+      getSnapshotDetails: async snapshotId => {
+        const res = await fetchDataRepo(`${root}/${snapshotId}`, _.merge(authOpts(), { signal }))
+        return res.json()
+      }
+    }
+  }
+})
```

Comment:
I might restructure this like so, to make it more generic for future data repo uses (you'd have to change it in other places too, of course):
```js
const DataRepo = signal => ({
  snapshots: () => {
    const root = `repository/v1/snapshots`

    return {
      snapshot: snapshotId => {

        return {
          details: async () => {
            const res = await fetchDataRepo(`${root}/${snapshotId}`, _.merge(authOpts(), { signal }))
            return res.json()
          }
        }
      }
    }
  }
})
```